colony_picker/inverse_kinematics.py

```diff
@@ -0,0 +1,634 @@
+            system being drawn, coordinate systems with names can be updated.
+            Defaults to "".
+    """
+    colors = ["red", "green", "blue"]
+    position = t_mat[:3, 3]
+    sphere = pv.Sphere(center=position, radius=4)
+    if base:
+        sphere_color = "black"
+    else:
+        sphere_color = "yellow"
+    plotter.add_mesh(sphere, color=sphere_color, name=(name + "_sphere"))
+    for axis_idx in range(3):
+        axis = pv.Arrow(
+            start=position, direction=t_mat[:3, axis_idx], scale=50)
+        plotter.add_mesh(axis, color=colors[axis_idx], name=(
+            name + f"_coord_{axis_idx}"))
+
+
+def draw_links(plotter: pv.Plotter, joint_positions: np.ndarray):
+    """Draws the backbone of the robot arm on plotter given a set of joint
+    positions relative to the base of the arm. 
+
+    Args:
+        plotter (pv.Plotter): the pyvista plotter object to draw the robot arm
+            on
+        joint_positions (np.ndarray): an n x 3 array denoting the position of 
+            each joint relative to the base of the robot in the format 
+            [[x 1, y 1, z 1], [x 2, y 2, z 2], ... , [x n, y n, z n]] where
+            n = number of joints + 1.
+    """
+    poly = pv.PolyData()
+    poly.points = joint_positions
+    cells = np.full((len(joint_positions)-1, 3), 2, dtype=np.int_)
+    cells[:, 1] = np.arange(0, len(joint_positions)-1, dtype=np.int_)
+    cells[:, 2] = np.arange(1, len(joint_positions), dtype=np.int_)
+    poly.lines = cells
+    poly["scalars"] = np.arange(poly.n_points)
+    tube = poly.tube(radius=10)
+    plotter.add_mesh(tube, color="black", name="robot skeleton", opacity=0.5)
+
+
+def draw_robot_arm(thetas: np.ndarray, p: pv.Plotter, dh_params: np.ndarray):
+    """Draws the robot arm on the plotter, with a coordinate system representing
+    each joint and a robot backbone connecting each joint. 
+
+    Args:
+        thetas (np.ndarray): a 1 x n array with joint angles in the format
+            [joint angle 1, joint angle 2, ... , joint angle n], where n is the
+            number of joints. 
+        p (pv.Plotter): the plotter object to draw the robot arm on.
+        dh_params (np.ndarray): a 4 x n array containing the dh parameters to 
+            describe every joint on the robot arm in the format
+            [[theta offset 1, theta offset 2, ... , theta offset n],
+            [alpha 1, alpha 2, ... , alpha n],
+            [a 1, a 2, ... , a n], 
+            [d 1, d 2, ... , d n]]
+            where n is the number of joints on the robot arm.
+    """
+    num_joints = len(thetas)
+    joint_positions = np.zeros((num_joints + 1, 3))
+    t_mats = get_t_mats(thetas, dh_params)
+    for i, t_mat in enumerate(t_mats):
+        joint_positions[i + 1, :3] = t_mat[:3, 3]
+        draw_coordinate_system(p, t_mat, name=f"theta{i}")
+    draw_links(p, joint_positions)
+
+
+def convert_pose_to_quat(pose_in_euler: np.ndarray):
```

Comment:
Maybe rename to `euler_pose_to_quat`, and just call `pose_to_euler` the name `pose`, and have it's value implied by the function name, and clarified in the argument doc string?